File: src/PoeLogsParser/Models/LogEntry.cs
using System;
using System.Collections;
using System.Collections.Generic;
using PoeLogsParser.Enums;
using PoeLogsParser.Models.Abstractions;

namespace PoeLogsParser.Models
{
    public class LogEntry : ILogEntry
    {
        public string Raw { get; set; }
        public DateTime Time { get; set; }
        public IEnumerable<LogEntryType> Types { get; set; }
        public LogEntryTag Tag { get; set; }

        public LogEntry(string raw, DateTime time, IEnumerable<LogEntryType> types,
            LogEntryTag tag = LogEntryTag.Info)
        {
            Raw = raw;
            Time = time;
            Types = types;
            Tag = tag;
        }
    }
}

File: ts/src/models/entries/LogEntry.ts
import { LogEntryTag } from "../../enums/LogEntryTag";
import { LogEntryType } from "../../enums/LogEntryType";

export class LogEntry {
	public raw!: String;
	public time!: Date;
	public types!: LogEntryType[];
	public tag!: LogEntryTag;

	constructor(raw: string, time: Date, types: LogEntryType[], tag: LogEntryTag) {
		this.raw = raw;
		this.time = time;
		this.types = types;
		this.tag = tag;
	}
};